src/deploy/functions/build.ts

```diff
@@ -251,32 +275,30 @@ export function resolveBackend(build: Build): backend.Backend {
 
       const bkEndpoint: backend.Endpoint = {
         id: endpointId,
-        project: "",
+        project: endpoint.project,
         region: region,
         entryPoint: endpoint.entryPoint,
         platform: endpoint.platform,
-        runtime: "",
-        labels: endpoint.labels,
-        environmentVariables: endpoint.environmentVariables,
-        secretEnvironmentVariables: undefined,
-        availableMemoryMb: endpoint.availableMemoryMb,
+        runtime: endpoint.runtime,
         timeoutSeconds: timeout,
         ...trigger,
       };
       proto.renameIfPresent(bkEndpoint, endpoint, "maxInstances", "maxInstances", resolveInt);
       proto.renameIfPresent(bkEndpoint, endpoint, "minInstances", "minInstances", resolveInt);
       proto.renameIfPresent(bkEndpoint, endpoint, "concurrency", "concurrency", resolveInt);
       proto.copyIfPresent(bkEndpoint, endpoint, "ingressSettings");
+      proto.copyIfPresent(bkEndpoint, endpoint, "availableMemoryMb");
+      proto.copyIfPresent(bkEndpoint, endpoint, "environmentVariables");
+      proto.copyIfPresent(bkEndpoint, endpoint, "labels");
+      // proto.copyIfPresent(bkEndpoint, endpoint, "secretEnvironmentVariables");
       if (endpoint.vpc) {
         bkEndpoint.vpc = {
-          connector: resolveString(endpoint.vpc.connector),
-          egressSettings: endpoint.vpc.egressSettings,
+          connector: resolveString(endpoint.vpc.connector).replace("$REGION", region),
```

Comment:
is "$REGION" actually a thing in this input string?

Reply:
It is, because of the difficulty of unrolling region in VPC connector strings discussed in chat. I'll add a comment to remind me of the reasoning here.